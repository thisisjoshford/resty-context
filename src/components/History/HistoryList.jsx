import React from 'react';
import PropTypes from 'prop-types';
import HistoryItem from './HistoryItem';

const HistoryList = ({ history }) => {
  const historyNodes = history
    .map((item, i) => (
      <li key={i}>
        <HistoryItem {...item} />
      </li>
    ));
  
  return (
    <ul>
      {historyNodes}
    </ul>
  );
};

HistoryList.propTypes = {
  history: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string.isRequired,
    method: PropTypes.string.isRequired
  })).isRequired
};

export default HistoryList;
