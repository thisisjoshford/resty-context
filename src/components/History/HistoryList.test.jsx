import React from 'react';
import { shallow } from 'enzyme';
import HistoryList from './HistoryList';

describe('History list component', () => {
  it('renders History list', () => {
    const historyArr = [{
      url: 'http://randomapi.com',
      method: 'GET'
    }];

    const wrapper = shallow(
      <HistoryList
        history={historyArr}
      />);
    expect(wrapper).toMatchSnapshot();
  });
});
