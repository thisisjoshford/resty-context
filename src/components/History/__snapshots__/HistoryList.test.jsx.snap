// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History list component renders History list 1`] = `
<ul>
  <li
    key="0"
  >
    <HistoryItem
      method="GET"
      url="http://randomapi.com"
    />
  </li>
</ul>
`;
